rdmd_test.d

```diff
@@ -48,42 +48,38 @@ else
 
 bool verbose = false;
 
-void main(string[] args)
+int main(string[] args)
 {
-    string rdmd; // path to rdmd executable
-    string defaultCompiler; // name of default compiler expected by rdmd
+    string compilers = "dmd"; // e.g. "ldmd2,gdmd" (comma-separated list of compiler names)
     bool concurrencyTest;
     string model = "64"; // build architecture for dmd
-    string testCompilerList; // e.g. "ldmd2,gdmd" (comma-separated list of compiler names)
 
     auto helpInfo = getopt(args,
-        "rdmd", "[REQUIRED] path to rdmd executable to test", &rdmd,
-        "rdmd-default-compiler", "[REQUIRED] default D compiler used by rdmd executable", &defaultCompiler,
+        "compilers", "comma-separated list of D compilers to test rdmd with (defaults to 'dmd')", &compilers,
         "concurrency", "whether to perform the concurrency test cases", &concurrencyTest,
         "m|model", "architecture to run the tests for [32 or 64]", &model,
-        "test-compilers", "comma-separated list of D compilers to test with rdmd", &testCompilerList,
         "v|verbose", "verbose output", &verbose,
     );
 
     void reportHelp(string errorMsg = null, string file = __FILE__, size_t line = __LINE__)
     {
         defaultGetoptPrinter("rdmd_test: a test suite for rdmd\n\n" ~
-                             "USAGE:\trdmd_test [OPTIONS]\n",
+                             "USAGE:\trdmd_test [OPTIONS] <rdmd-binary>\n",
                              helpInfo.options);
         enforce(errorMsg is null, errorMsg, file, line);
     }
 
-    if (helpInfo.helpWanted)
+    if (args.length == 1 || helpInfo.helpWanted)
```

Comment:
The various arguments-validation checks introduced here and below look harder to understand and maintain than just preserving the `--rdmd` flag.  I'm not sure that the simplicity of being able to type

```
rdmd_test [PATH_TO_RDMD]
```

... is worth it when it's simple enough to type

```
rdmd_test --rdmd [PATH_TO_RDMD]
```

... especially bearing in mind that the makefile design is set up to avoid anyone having to run `rdmd_test` directly.